internal/client/client_fail_test.go

```diff
@@ -59,9 +59,8 @@ func TestPanicInRecvLoop(t *testing.T) {
 	defer server.Stop()
 
 	addr := fmt.Sprintf("%s:%d", "127.0.0.1", port)
-	rpcClient := NewRPCClient(config.Security{}, func(c *RPCClient) {
-		c.dialTimeout = time.Second / 3
-	})
+	rpcClient := NewRPCClient()
+	rpcClient.dialTimeout = time.Second / 3
```

Comment:
I'm a bit curious about the `opts` design. If the options can be set by this way, why need we pass opts in `NewRPCClient`?

Reply:
opts can be used by library users that do not have access to private fields. For `dialTimeout`, we only need to update it in the same package (at least for now), so don't have to use opts.

Reply:
Ah, I see. If any field that needs to be modified by the library user, we provide a function returning `Opt` like `WithSecurity`.